nixos/modules/services/networking/vpp.nix

```diff
@@ -0,0 +1,525 @@
+{
+  config,
+  lib,
+  pkgs,
+  ...
+}:
+let
+  inherit (lib)
```

Comment:
You could probably slim down on some of the other `types.`/`lib.` usages by adding them to an inherit - it makes it much easier to read if it's just a small suffix rather than a lot of repetition across options. I'm thinking primarily stuff like `types.str`, `types.nullOr` etc. They can be replaced with:
```nix
let
  inherit (lib.types)
    str
    nullOr
    ;
in
```
at the top.

Reply:
This module uses a lot of different types, so the list would need to be pretty large and clutter the top of the file more than I'd like:
```nix
let
  inherit (lib.types)
    int
    ints
    str
    lines
    path
    bool
    submodule
    attrsOf
    nullOr
    oneOf
    listOf
    ;
in
```
It also uses a few custom `semanticTypes`, and declaring `types` explicitly makes it unambiguous where everything comes from.

Reply:
I get where you're coming from, I'm just speaking from my experience which may be different. Ultimately I think it's your choice, but if you're interested to see what the difference is feel free to check out https://github.com/NixOS/nixpkgs/blob/master/nixos%2Fmodules%2Fservices%2Fsecurity%2Fkanidm.nix. It's a much more extreme example, but I think it's better for it.